room/compiler-processing/src/main/java/androidx/room/compiler/processing/ksp/KSTypeExt.kt

```diff
@@ -61,8 +61,20 @@ private fun KSName.typeName(): ClassName? {
 }
 
 private fun KSDeclaration.typeName(): ClassName? {
-    // we are only interested in qualified name as if it does not exist, it is an error for Room
-    return qualifiedName?.typeName()
+    // if there is no qualified name, it is an error for room
```

Comment:
I am planning to let KSType.declaration to return `null` for error types, so you won't need to rely on this behavior, a local declaration does not have a qualified name, Room might not be concerned though.

Reply:
yea i don't think we have a use case where room would generate code that would access local declarations since we generate code in another java class. at least, not for near future